src/pathfinder/water_regions.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file water_regions.h Handles dividing the water in the map into regions to assist pathfinding. */
+
+#ifndef WATER_REGIONS_H
+#define WATER_REGIONS_H
+
+#include "tile_type.h"
+#include "map_func.h"
+
+#include <vector>
+#include <functional>
+
+using TWaterRegionPatchLabel = uint8;
+using TWaterRegionIndex = uint;
```

Comment:
```suggestion
using TWaterRegionPatchLabel = uint8_t;
using TWaterRegionIndex = uint;
```
We'll be going to the std types, so `uint8_t`.